Missing commons-lang3 runtime dependency causes NoClassDefFoundError in DefaultLoadBalancingPolicy (4.19.0.5)

## Bug Report

### Version

`com.scylladb:java-driver-core:4.19.0.5`

### Description

Starting with version `4.19.0.5`, `DefaultLoadBalancingPolicy.moveReplicasToFront()` uses `org.apache.commons.lang3.tuple.Pair`, but `commons-lang3` is **not** declared as a dependency in `core/pom.xml` (or the parent POM). This causes a `NoClassDefFoundError` / `ClassNotFoundException` at runtime when the load balancing policy attempts to build a query plan for regular (non-LWT) requests on multi-node clusters.

### Root Cause

PR [#784](https://github.com/scylladb/java-driver/pull/784) (*"Fix: LWT routing enhancements to utilize LBP"*), merged on 2026-01-27, refactored `DefaultLoadBalancingPolicy` to return `Pair<Integer, Integer>` from the `moveReplicasToFront` method:

```java
import org.apache.commons.lang3.tuple.Pair;
// ...
private Pair<Integer, Integer> moveReplicasToFront(Object[] currentNodes, List<Node> allReplicas) {
    // ...
    return Pair.of(replicaCount, localRackReplicaCount);
}
```

However, the corresponding `org.apache.commons:commons-lang3` dependency was never added to [`core/pom.xml`](https://github.com/scylladb/java-driver/blob/scylla-4.x/core/pom.xml) or the parent [`pom.xml`](https://github.com/scylladb/java-driver/blob/scylla-4.x/pom.xml).

### Why this is not caught during testing

The failure only occurs in multi-node production clusters where `DefaultLoadBalancingPolicy.newQueryPlanRegular()` is invoked and replicas are non-empty. Single-node integration tests don't exercise the `moveReplicasToFront` code path because there are no replicas to move, so the `Pair` class is never loaded.

### Stack Trace

```
java.lang.NoClassDefFoundError: org/apache/commons/lang3/tuple/Pair
    at com.datastax.oss.driver.internal.core.loadbalancing.DefaultLoadBalancingPolicy.moveReplicasToFront(DefaultLoadBalancingPolicy.java:268)
    at com.datastax.oss.driver.internal.core.loadbalancing.DefaultLoadBalancingPolicy.newQueryPlanRegular(DefaultLoadBalancingPolicy.java:205)
    at com.datastax.oss.driver.internal.core.loadbalancing.DefaultLoadBalancingPolicy.newQueryPlan(DefaultLoadBalancingPolicy.java:175)
    at com.datastax.oss.driver.internal.core.metadata.LoadBalancingPolicyWrapper.newQueryPlan(LoadBalancingPolicyWrapper.java:159)
    at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.onThrottleReady(CqlRequestHandler.java:213)
    at com.datastax.oss.driver.internal.core.session.throttling.PassThroughRequestThrottler.register(PassThroughRequestThrottler.java:54)
    at com.datastax.oss.driver.internal.core.cql.CqlRequestHandler.<init>(CqlRequestHandler.java:191)
    at com.datastax.oss.driver.internal.core.cql.CqlRequestAsyncProcessor.process(CqlRequestAsyncProcessor.java:46)
    at com.datastax.oss.driver.internal.core.cql.CqlRequestAsyncProcessor.process(CqlRequestAsyncProcessor.java:31)
    at com.datastax.oss.driver.internal.core.session.DefaultSession.execute(DefaultSession.java:241)
    ...
Caused by: java.lang.ClassNotFoundException: org.apache.commons.lang3.tuple.Pair
    ... 38 common frames omitted
```

### Suggested Fix

Either:

1. **Add `commons-lang3` as a runtime dependency** in `core/pom.xml`:

```xml
<dependency>
  <groupId>org.apache.commons</groupId>
  <artifactId>commons-lang3</artifactId>
  <version>3.17.0</version>
</dependency>
```

2. **Or replace the usage** with a simple two-element return type (e.g., an `int[]` or a private inner record/class) to avoid pulling in an extra transitive dependency for a single usage.

### Workaround

Users can add `commons-lang3` explicitly to their own project dependencies:

**Maven:**
```xml
<dependency>
  <groupId>org.apache.commons</groupId>
  <artifactId>commons-lang3</artifactId>
  <version>3.17.0</version>
</dependency>
```

**sbt:**
```scala
libraryDependencies += "org.apache.commons" % "commons-lang3" % "3.17.0"
```

### Environment

- ScyllaDB Java Driver: `4.19.0.5`
- Java: 17+
- Multi-node ScyllaDB cluster (not reproducible with single-node test setups)

--- a/core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicy.java
+++ b/core/src/main/java/com/datastax/oss/driver/internal/core/loadbalancing/DefaultLoadBalancingPolicy.java
@@ -51,7 +51,6 @@ import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.atomic.AtomicLongArray;
 import net.jcip.annotations.ThreadSafe;
-import org.apache.commons.lang3.tuple.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -202,10 +201,10 @@ public class DefaultLoadBalancingPolicy extends BasicLoadBalancingPolicy impleme
     Object[] currentNodes = getLiveNodes().dc(getLocalDatacenter()).toArray();
     int replicaCount = 0; // in currentNodes
     if (!replicas.isEmpty()) {
-      Pair<Integer, Integer> counts = moveReplicasToFront(currentNodes, replicas);
-      replicaCount = counts.getLeft();
+      int[] counts = moveReplicasToFront(currentNodes, replicas);
+      replicaCount = counts[0];
 
-      int localRackReplicaCount = counts.getRight(); // in currentNodes
+      int localRackReplicaCount = counts[1]; // in currentNodes
 
       if (replicaCount > 1) {
         shuffleLocalRackReplicasAndReplicas(currentNodes, replicaCount, localRackReplicaCount);
@@ -249,8 +248,7 @@ public class DefaultLoadBalancingPolicy extends BasicLoadBalancingPolicy impleme
     return orderedReplicas;
   }
 
-  private Pair<Integer, Integer> moveReplicasToFront(
-      Object[] currentNodes, List<Node> allReplicas) {
+  private int[] moveReplicasToFront(Object[] currentNodes, List<Node> allReplicas) {
     int replicaCount = 0, localRackReplicaCount = 0;
     for (int i = 0; i < currentNodes.length; i++) {
       Node node = (Node) currentNodes[i];
@@ -265,7 +263,7 @@ public class DefaultLoadBalancingPolicy extends BasicLoadBalancingPolicy impleme
         replicaCount++;
       }
     }
-    return Pair.of(replicaCount, localRackReplicaCount);
+    return new int[] {replicaCount, localRackReplicaCount};
   }
 
   private void shuffleLocalRackReplicasAndReplicas(

--- a/core/src/test/java/com/datastax/oss/driver/internal/core/type/codec/VectorCodecTest.java
+++ b/core/src/test/java/com/datastax/oss/driver/internal/core/type/codec/VectorCodecTest.java
@@ -34,8 +34,8 @@ import com.tngtech.java.junit.dataprovider.DataProviderRunner;
 import com.tngtech.java.junit.dataprovider.UseDataProvider;
 import java.nio.ByteBuffer;
 import java.time.LocalTime;
+import java.util.Arrays;
 import java.util.HashMap;
-import org.apache.commons.lang3.ArrayUtils;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 
@@ -128,7 +128,10 @@ public class VectorCodecTest {
   @Test
   @UseDataProvider("dataProvider")
   public void should_throw_on_encode_with_too_many_elements(TestDataContainer testData) {
-    Object[] doubled = ArrayUtils.addAll(testData.getValues(), testData.getValues());
+    Object[] first = testData.getValues();
+    Object[] second = testData.getValues();
+    Object[] doubled = Arrays.copyOf(first, first.length + second.length);
+    System.arraycopy(second, 0, doubled, first.length, second.length);
     TypeCodec<CqlVector<Object>> codec = getCodec(testData.getDataType());
     assertThatThrownBy(() -> codec.encode(CqlVector.newInstance(doubled), ProtocolVersion.DEFAULT))
         .isInstanceOf(IllegalArgumentException.class);

--- a/pom.xml
+++ b/pom.xml
@@ -100,7 +100,7 @@
     <skipUnitTests>${skipTests}</skipUnitTests>
     <release.autopublish>false</release.autopublish>
     <pushChanges>false</pushChanges>
-    <mockitoopens.argline />
+    <mockitoopens.argline/>
   </properties>
   <dependencyManagement>
     <dependencies>

